core/src/main/java/org/apache/accumulo/core/iterators/conf/ColumnSet.java

```diff
@@ -146,10 +146,13 @@ static Text decode(String s) {
       if (sb[i] != '%') {
         t.append(new byte[] {sb[i]}, 0, 1);
       } else {
-        byte hex[] = {sb[++i], sb[++i]};
-        String hs = new String(hex, UTF_8);
-        int b = Integer.parseInt(hs, 16);
-        t.append(new byte[] {(byte) b}, 0, 1);
+        i++;
```

Comment:
You change doesn't look equivalent. It looks like you should have something like below

```java
        int x = ++i;
        int y = ++i;
        if (y < sb.length) {
          byte hex[] = {sb[x], sb[y]};
          String hs = new String(hex, UTF_8);
          int b = Integer.parseInt(hs, 16);
          t.append(new byte[]{(byte) b}, 0, 1);
        }
```

Reply:
Oooo yeah good catch.

Reply:
I should add a check for both indexes as well.

Reply:
Also should throw an exception on the else.  Old code would have thrown an array index out of bounds exception.  Can throw something with a nicer message.  I think it always expects to see two chars after `%`

Reply:
i don't think you need to. if y is less than `sb.length` than x should be too